lib/openstudio-standards/weather/stat_file.rb

```diff
@@ -284,13 +284,20 @@ def parse
         degree_day_info.each { |dd_info| parse_dd_info(dd_info) }
 
         # parse design temperatures
+        # TMYx stat file format is different than TMY3, cf #1948
+        is_tmyx = @text.include?("Wind Shelter Factor")
         temperature_info = [
-          { name: 'Heating Design Temperatures', regex: /Heating(\s*\d+.*)\n/, container: @heating_design_info, size: 15 },
+          { name: 'Heating Design Temperatures', regex: /Heating(\s*\d+.*)\n/, container: @heating_design_info, size: is_tmyx ? 16 : 15 },
           { name: 'Cooling Design Temperatures', regex: /Cooling(\s*\d+.*)\n/, container: @cooling_design_info, size: 32 },
-          { name: 'Extreme Design Temperatures', regex: /\s*Extremes\s*(.*)\n/, container: @extremes_design_info, size: 16 },
+          { name: 'Extreme Design Temperatures', regex: /\s*Extremes\s*(.*)\n/, container: @extremes_design_info, size: is_tmyx ? 15 : 16 },
           { name: 'Monthly Dry Bulb Temperatures', regex: /Daily Avg(.*)\n/, container: @monthly_dry_bulb, size: 12 }
         ]
         temperature_info.each { |temp_info| parse_design_temp_info(temp_info) }
+        if is_tmyx
+          @heating_design_info.pop() # remove the last value, which is the Wind Shelter Factor
+          # There is no WBmax in TMYx files, so add a nil
+          @extremes_design_info.insert(3, nil)
+        end
```

Comment:
Then we fix it up.

I don't think ANYONE is using these arrays to be honest (they are awful to work with because they have no header information), but still make it consistent so the arrays are similar to what you'd get with a TMY3 stat file.

Reply:
yeah, this is archaic code that does text parsing of a .stat file. I'd like to just get rid of the .stat file entirely if possible and just calculate directly from the .epw.